extension/js/common/api/email-provider/sendable-msg.ts

```diff
@@ -27,7 +27,48 @@ export class SendableMsg {
 
   public sign?: (signable: string) => Promise<string>;
 
-  public static create = async (acctEmail: string, { from, recipients, subject, body, atts, thread, type, isDraft }: SendableMsgDefinition): Promise<SendableMsg> => {
+  //  return await SendableMsg.create(this.acctEmail, { this.headers(newMsg), body, type: 'smimeEncrypted', atts: [], isDraft: this.isDraft });
```

Comment:
All of these method signatures should be completely revised else there's very little benefit in refactoring them.

So that I can use S/mime like follows:
```
const msg = SendableMsg.createSMime(acctEmail, this.headers(newMsg), Buf.fromUint8(encryptedMessage.data))
```

The usage of each of these methods has to be very specific and clear. They can differ from one another a bit in how they are used. As it is now, they all look about the same, accepting a large bundle of arguments mixed together, so the usage is not much clearer from before.

Reply:
ok, makes sense. I'll try put more abstraction on them.

Reply:
@tomholub , pls. check again. The names are renamed and i think the abstraction over methods is much better. Actually for the latter I just created a basic signature:
```
async (
     acctEmail: string,  
     headers: SendableMsgHeaders, 
     body: SendableMsgBody, 
     attachments: Att[], 
    options: SendableMsgOptions
): Promise<SendableMsg>;
```
.. and based on (new) method name the inner implementation may change a bit per case. Also for a couple of methods the options (new custom type combining the original type and isDraft parameters) parameter may be optional.

Reply:
Note:.. when we are ready i'll remove the numerous inline comments ..